Lib/ast.py

```diff
@@ -1053,15 +1055,80 @@ def visit_AsyncWith(self, node):
         with self.block(extra=self.get_type_comment(node)):
             self.traverse(node.body)
 
+    def _str_literal_helper(
+        self, value, quote_types=("'", '"', '"""', "'''"), escape=""
+    ):
+        """Helper for writing string literals, minimising escapes.
+        Returns (possible quote types, string literal to write).
+        """
+        # Escape characters we've been told to escape and any non-printable
+        # characters. The logic for determining non-printable characters is
+        # based on that in Tools/unicode/makeunicodedata.py
+        escape = set(escape) | {
+            c for c in value
+            if c not in ' \n\t' and unicodedata.category(c)[0] in ("C", "Z")
+        }
+        qts = quote_types
+        val = value.replace("\\", "\\\\")
+        for c in escape:
+            val = val.replace(c, c.encode('unicode_escape').decode('ascii'))
```

Comment:
I don't think replacing is the suitable solution for this, you should use `var[x] = blabla` or simply `val = "".join(char.encode(x).decode(y) for char in val).replace("\\", "\\\\")`...

Reply:
Changed to use join! We need to replace backslashes before escaping, though.